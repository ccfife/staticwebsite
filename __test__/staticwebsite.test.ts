import {SynthUtils} from '@aws-cdk/assert';
import {Stack, App} from '@aws-cdk/core';
import '@aws-cdk/assert/jest';

import website = require('../lib/staticwebsite-stack');
import { tsIndexSignature } from '@babel/types';

test('static website infra has not changed', () => {
    const app = new App();
    const site = new website.StaticwebsiteStack(app, 'testwebsite');
    expect(SynthUtils.toCloudFormation(site)).toMatchSnapshot();

});

test('check S3 bucket website configuration', () => {
     const app = new App();
     const site = new website.StaticwebsiteStack(app, 'testwebsite');

     expect(site).toHaveResource('AWS::S3::Bucket', {
         WebsiteConfiguration: {
            ErrorDocument: "404.html", 
            IndexDocument: "index.html"
         }
     });
});

test('Check CloudFront origin path config', () => {
    const app = new App();
    const site = new website.StaticwebsiteStack(app, 'testwebsite');

    expect(site).toHaveResourceLike('AWS::CloudFront::Distribution', {
        "DistributionConfig": {
            "DefaultRootObject": "index.html",
            "Origins": [
                {
                    "OriginPath": "/web/static"
                }
            ]
        }
    });
});