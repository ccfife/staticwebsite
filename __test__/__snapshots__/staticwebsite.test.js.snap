// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`static website infra has not changed 1`] = `
Object {
  "Outputs": Object {
    "CloudFrontURL": Object {
      "Description": "CDN URL",
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Fn::GetAtt": Array [
                "cloudfrontCFDistribution0FC24C70",
                "DomainName",
              ],
            },
          ],
        ],
      },
    },
  },
  "Resources": Object {
    "BucketOrigin": Object {
      "Properties": Object {
        "CloudFrontOriginAccessIdentityConfig": Object {
          "Comment": "sonar master",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "CDKpipeline2E0BE889": Object {
      "DependsOn": Array [
        "CDKpipelineRoleDefaultPolicy6C05FD8D",
        "CDKpipelineRoleB4F4E9E9",
      ],
      "Properties": Object {
        "ArtifactStore": Object {
          "EncryptionKey": Object {
            "Id": Object {
              "Fn::GetAtt": Array [
                "CDKpipelineArtifactsBucketEncryptionKeyF46454F3",
                "Arn",
              ],
            },
            "Type": "KMS",
          },
          "Location": Object {
            "Ref": "CDKpipelineArtifactsBucket41E5DE9A",
          },
          "Type": "S3",
        },
        "Name": "SonarMasterPipeline",
        "RoleArn": Object {
          "Fn::GetAtt": Array [
            "CDKpipelineRoleB4F4E9E9",
            "Arn",
          ],
        },
        "Stages": Array [
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Source",
                  "Owner": "ThirdParty",
                  "Provider": "GitHub",
                  "Version": "1",
                },
                "Configuration": Object {
                  "Branch": "master",
                  "OAuthToken": "{{resolve:secretsmanager:ccfife_github:SecretString:GitHub::}}",
                  "Owner": "ccfife",
                  "PollForSourceChanges": false,
                  "Repo": "staticwebsite",
                },
                "InputArtifacts": Array [],
                "Name": "GitHub_Source",
                "OutputArtifacts": Array [
                  Object {
                    "Name": "Artifact_Source_GitHub_Source",
                  },
                ],
                "RunOrder": 1,
              },
            ],
            "Name": "Source",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Deploy",
                  "Owner": "AWS",
                  "Provider": "S3",
                  "Version": "1",
                },
                "Configuration": Object {
                  "BucketName": Object {
                    "Ref": "staticWebsitev1D351E58C",
                  },
                  "Extract": "true",
                },
                "InputArtifacts": Array [
                  Object {
                    "Name": "Artifact_Source_GitHub_Source",
                  },
                ],
                "Name": "S3Deploy",
                "OutputArtifacts": Array [],
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "CDKpipelineDeployS3DeployCodePipelineActionRole7D14C6B4",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Deploy",
          },
        ],
      },
      "Type": "AWS::CodePipeline::Pipeline",
    },
    "CDKpipelineArtifactsBucket41E5DE9A": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "KMSMasterKeyID": Object {
                  "Fn::GetAtt": Array [
                    "CDKpipelineArtifactsBucketEncryptionKeyF46454F3",
                    "Arn",
                  ],
                },
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "CDKpipelineArtifactsBucketEncryptionKeyF46454F3": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "KeyPolicy": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "kms:Create*",
                "kms:Describe*",
                "kms:Enable*",
                "kms:List*",
                "kms:Put*",
                "kms:Update*",
                "kms:Revoke*",
                "kms:Disable*",
                "kms:Get*",
                "kms:Delete*",
                "kms:ScheduleKeyDeletion",
                "kms:CancelKeyDeletion",
                "kms:GenerateDataKey",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "CDKpipelineRoleB4F4E9E9",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "CDKpipelineDeployS3DeployCodePipelineActionRole7D14C6B4",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "CDKpipelineDeployS3DeployCodePipelineActionRole7D14C6B4": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "CDKpipelineDeployS3DeployCodePipelineActionRoleDefaultPolicy97623ACC": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "staticWebsitev1D351E58C",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "staticWebsitev1D351E58C",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "CDKpipelineArtifactsBucket41E5DE9A",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "CDKpipelineArtifactsBucket41E5DE9A",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "CDKpipelineArtifactsBucketEncryptionKeyF46454F3",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CDKpipelineDeployS3DeployCodePipelineActionRoleDefaultPolicy97623ACC",
        "Roles": Array [
          Object {
            "Ref": "CDKpipelineDeployS3DeployCodePipelineActionRole7D14C6B4",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CDKpipelineRoleB4F4E9E9": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "codepipeline.",
                      Object {
                        "Ref": "AWS::URLSuffix",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "CDKpipelineRoleDefaultPolicy6C05FD8D": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "CDKpipelineArtifactsBucket41E5DE9A",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "CDKpipelineArtifactsBucket41E5DE9A",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "CDKpipelineArtifactsBucketEncryptionKeyF46454F3",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "CDKpipelineDeployS3DeployCodePipelineActionRole7D14C6B4",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CDKpipelineRoleDefaultPolicy6C05FD8D",
        "Roles": Array [
          Object {
            "Ref": "CDKpipelineRoleB4F4E9E9",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CDKpipelineSourceGitHubSourceWebhookResource2DBA157F": Object {
      "Properties": Object {
        "Authentication": "GITHUB_HMAC",
        "AuthenticationConfiguration": Object {
          "SecretToken": "{{resolve:secretsmanager:ccfife_github:SecretString:GitHub::}}",
        },
        "Filters": Array [
          Object {
            "JsonPath": "$.ref",
            "MatchEquals": "refs/heads/{Branch}",
          },
        ],
        "RegisterWithThirdParty": true,
        "TargetAction": "GitHub_Source",
        "TargetPipeline": Object {
          "Ref": "CDKpipeline2E0BE889",
        },
        "TargetPipelineVersion": 1,
      },
      "Type": "AWS::CodePipeline::Webhook",
    },
    "cloudfrontCFDistribution0FC24C70": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "CacheBehaviors": Array [],
          "DefaultCacheBehavior": Object {
            "AllowedMethods": Array [
              "GET",
              "HEAD",
              "OPTIONS",
            ],
            "CachedMethods": Array [
              "GET",
              "HEAD",
            ],
            "ForwardedValues": Object {
              "Cookies": Object {
                "Forward": "none",
              },
              "QueryString": false,
            },
            "TargetOriginId": "origin1",
            "ViewerProtocolPolicy": "allow-all",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": Array [
            Object {
              "DomainName": Object {
                "Fn::GetAtt": Array [
                  "staticWebsitev1D351E58C",
                  "RegionalDomainName",
                ],
              },
              "Id": "origin1",
              "OriginPath": "/web/static",
              "S3OriginConfig": Object {
                "OriginAccessIdentity": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "origin-access-identity/cloudfront/",
                      Object {
                        "Ref": "BucketOrigin",
                      },
                    ],
                  ],
                },
              },
            },
          ],
          "PriceClass": "PriceClass_All",
          "ViewerCertificate": Object {
            "CloudFrontDefaultCertificate": true,
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "staticWebsitev1D351E58C": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "WebsiteConfiguration": Object {
          "ErrorDocument": "404.html",
          "IndexDocument": "index.html",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "staticWebsitev1PolicyF4B9EE9E": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "staticWebsitev1D351E58C",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "CanonicalUser": Object {
                  "Fn::GetAtt": Array [
                    "BucketOrigin",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "staticWebsitev1D351E58C",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "staticWebsitev1D351E58C",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
}
`;
